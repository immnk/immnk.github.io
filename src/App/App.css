.root {
  position: relative;
}

.container {
  position: relative;
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2.4rem;
  text-align: center;
  color: #bfacaa;
  background-color: #f4f1bb;
}

.header {
  font-size: 5rem;
  font-family: 'Abel', sans-serif;
}

.highlight {
  color: #330c2f;
  font-family: 'Abel', sans-serif;
}

.header + p {
  padding: 1.2rem;
  color: #474448;
  font-size: 1.3rem;
}

.devImage {
  position: absolute;
  top: 5%;
  transform: translateX(-50%) translateY(-50%);
  left: 50%;
  width: 200px;
  height: 200px;
  opacity: 0.5;
  background: #f4f1bb url(./dev.png) no-repeat;
  background-size: contain;
  border-radius: 50%;
  z-index: 1;

  @media screen and (max-width: 540px) {
    top: 50px;
    transform: translateX(-50%);
  }
}

.iconsSocial {
  position: absolute;
  left: 50%;
  bottom: 0;
  transform: translateX(-50%) translateY(-50%);
  a {
    color: #bfacaa;
    :global {
      i {
        font-size: 3rem;
        padding: 1rem;
        &:hover {
          color: #474448;
        }
      }
    }
  }

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}
