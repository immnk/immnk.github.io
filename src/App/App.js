import css from "./App.module.css";
import "./App.css";

function App() {
  return (
    <div className={css.root}>
      <div className={css.devImage}></div>
      <div className={css.container}>
        <h1 className={css.header}>
          Hi, I'm <span className={css.highlight}>Mani ðŸ‘“</span>
        </h1>
        <p>
          Mobile and Front End Developer | UI UX | Web Designer | Foodie | Tech
          Enthusiastic | Developer #DBS | Proud Indian
        </p>
        <div className="iconsSocial">
          <a target="_blank" href="https://github.com/immnk" rel="noreferrer">
            <i className="fab fa-github"></i>
          </a>
          <a
            target="_blank"
            href="https://twitter.com/manimnk"
            rel="noreferrer"
          >
            <i className="fab fa-twitter"></i>
          </a>
          <a
            target="_blank"
            href="https://stackoverflow.com/story/manikantamnk"
            rel="noreferrer"
          >
            <i className="fab fa-stack-overflow"></i>
          </a>
          <a target="_blank" href="https://dev.to/immnk" rel="noreferrer">
            <i className="fab fa-dev"></i>
          </a>
          <a
            target="_blank"
            href="https://www.linkedin.com/in/manikantatankala"
            rel="noreferrer"
          >
            <i className="fab fa-linkedin"></i>
          </a>
        </div>
      </div>
    </div>
  );
}

export default App;
