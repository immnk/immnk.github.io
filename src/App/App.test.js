import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

jest.mock('react-ga');

const renderApp = (props) => shallow(<App {...props} />);

describe('Test App', () => {
  it('render app', () => {
    const actual = renderApp();
    expect(actual).toMatchSnapshot();
  });
});
