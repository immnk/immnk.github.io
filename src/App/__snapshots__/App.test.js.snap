// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App render app 1`] = `
<div
  className="root"
>
  <div
    className="devImage"
  />
  <div
    className="container"
  >
    <h1
      className="header"
    >
      Hi, I'm 
      <span
        className="highlight"
      >
        Mani ðŸ‘“
      </span>
    </h1>
    <p>
      Mobile and Front End Developer | UI UX | Web Developer | Foodie | Tech Enthusiastic | Developer #DBS #SGX #Sirius | Proud Indian
    </p>
    <div
      className="iconsSocial"
    >
      <a
        href="https://github.com/immnk"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-github"
        />
      </a>
      <a
        href="https://twitter.com/manimnk"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-twitter"
        />
      </a>
      <a
        href="https://stackoverflow.com/story/manikantamnk"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-stack-overflow"
        />
      </a>
      <a
        href="https://dev.to/immnk"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-dev"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/manikantatankala"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-linkedin"
        />
      </a>
    </div>
  </div>
  <AboutMe />
  <Roles />
  <Work />
  <Recommendations />
</div>
`;
